src/App.tsx

```diff
@@ -8,7 +7,58 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [select, setSelect] = useState('all');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [selectTodo, setSelectTodo] = useState<Todo>();
+
+  useEffect(() => {
+    getTodos().then(result => setAllTodos(result));
+
+    setVisibleTodos(allTodos.filter(todo => {
+      switch (select) {
+        case ('all'):
+          return todo.title.toUpperCase().includes(query.toUpperCase().trim());
```

Comment:
You use the same actions for all 3 cases here `todo.title.toUpperCase().includes(query.toUpperCase().trim())`. Move it to a variable.